controllers/qperf/serverdeployment.go

```diff
@@ -0,0 +1,120 @@
+/*
+Copyright 2019 The xridge kubestone contributors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package qperf
+
+import (
+	"fmt"
+	"strconv"
+
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	perfv1alpha1 "github.com/xridge/kubestone/api/v1alpha1"
+)
+
+// +kubebuilder:rbac:groups=apps,resources=deployments,verbs=get;list;create;delete;watch
+
+func serverDeploymentName(cr *perfv1alpha1.Qperf) string {
+	return cr.Name
+}
+
+// NewServerDeployment create a qperf server deployment from the
+// provided Qperf Benchmark Definition.
+func NewServerDeployment(cr *perfv1alpha1.Qperf) *appsv1.Deployment {
+	replicas := int32(1)
+
+	labels := map[string]string{
+		"kubestone.xridge.io/app":     "qperf",
+		"kubestone.xridge.io/cr-name": cr.Name,
+	}
+	// Let's be nice and don't mutate CRs label field
+	for k, v := range cr.Spec.ServerConfiguration.PodLabels {
+		labels[k] = v
+	}
+
+	qperfCmdLineArgs := []string{"--listen_port", strconv.Itoa(perfv1alpha1.QperfPort)}
+
+	// Qperf Server does not like if probe connections are made to the port,
+	// therefore we are checking if the port if open or not via awk script
+	// the solution does not assume to have netstat installed in the container
+	readinessAwkCmd := fmt.Sprintf("BEGIN{err=1}toupper($2)~/:%04X$/{err=0}END{exit err}", perfv1alpha1.QperfPort)
```

Comment:
This is nicer than the one in iperf. Could you please also change iperf to have similar check?
It might even make sense to move this to pkg/utils (or similar).

Reply:
Added to iperf3 and removed here.